### PR TITLE
migrate RNTesterActivity to Kotlin

### PR DESCRIPTION
**Pull request by BogiKay**
## Summary:

PR converts RNTesterActivity to Kotlin as requested in https://github.com/facebook/react-native/issues/38825 . 

## Changelog:

[INTERNAL] [CHANGED] - Migrate RNTesterActivity to Kotlin

## Test Plan:

1. run `yarn android`
2. Check whether RN Tester runs as expected

### CODE REVIEW

**Comment by BogiKay** [comment]
Hi @ryancat, can you share the results of an Internal linter? I'd like to fix issue that was found

File: packages/rn-tester/android/app/src/main/java/com/facebook/react/uiapp/RNTesterActivity.kt [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+import android.os.Bundle;
+import com.facebook.react.ReactActivity;
+import com.facebook.react.defaults.DefaultNewArchitectureEntryPoint.fabricEnabled
+import com.facebook.react.defaults.DefaultReactActivityDelegate;
+
+class RNTesterActivity : ReactActivity() {
+  class RNTesterActivityDelegate(val activity: ReactActivity, mainComponentName: String) : DefaultReactActivityDelegate(activity, mainComponentName, fabricEnabled) {
+    private val PARAM_ROUTE = "route"
+    private var initialProps: Bundle? = null
```
Comment (cortinico) [inline packages/rn-tester/android/app/src/main/java/com/facebook/react/uiapp/RNTesterActivity.kt:18]:
```suggestion
    private lateinit var initialProps: Bundle
```

File: packages/rn-tester/android/app/src/main/java/com/facebook/react/uiapp/RNTesterActivity.kt [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+      // Get remote param before calling super which uses it
+      val bundle = activity.getIntent()?.getExtras()
+
+      if (bundle != null && bundle.containsKey(PARAM_ROUTE)) {
+        val routeUri = with(StringBuilder("rntester://example/")) {
```
Comment (cortinico) [inline packages/rn-tester/android/app/src/main/java/com/facebook/react/uiapp/RNTesterActivity.kt:25]: Not sure we need a `StringBuilder` here. Let's just do string interpolation:
```
"rntester://example/${bundle.getString(PARAM_ROUTE)}Example"
```

**Review by cortinico — CHANGES_REQUESTED** [review]
Thanks for sending this over @BogiKay 
There are a couple of changes that needs to be applied before we can merge this one

**Pull request closed without merging**